bin/monitor

```diff
@@ -3,7 +3,7 @@
 
```

Comment:
re, commit message:

```

This approach is also more memory efficient. Memory efficiency is especially
important during high loads. Process forking at those scenarios can cause OOM
errors, thus causing us to unnecessarily declare unavailability.
```

I'd reword "efficiency" (which tends to be read mostly about size, where it ends up being somewhat better for lack of an extra shell/bash and `psql`) and perhaps itemize (much) reduced frequency of memory mapping in the operating system.  It is this memory mapping that can fail with ENOMEM. The difference in magnitude in amount of new memory maps is likely thousands to one (as simple queries in a monitoring loop will do a few small pallocs in the same virtual memory region over and over, not running `mmap` nor `brk`/`sbrk`), the largest magnitude effect by far.

I am not sure about the optimal wording here, you could say "memory allocations," but don't care about most (small) memory allocations e.g. palloc/malloc as such, since they tend to grab a bunch of virtual memory from the OS (which is the part that fails) and then do bookkeeping inside there. Perhaps while a bit ambiguous, it more quickly gets the reader's head in the right place.

Reply:
Updated the commit message.